src/lib/autocomplete/autocomplete-trigger.ts

```diff
@@ -523,13 +523,13 @@ export class MatAutocompleteTrigger implements ControlValueAccessor, OnDestroy {
       if (this._viewportRuler) {
         this._viewportSubscription = this._viewportRuler.change().subscribe(() => {
           if (this.panelOpen && this._overlayRef) {
-            this._overlayRef.updateSize({width: this._getHostWidth()});
+            this._overlayRef.updateSize({width: this._getWidth()});
           }
         });
       }
     } else {
       // Update the panel width and direction, in case anything has changed.
-      this._overlayRef.updateSize({width: this._getHostWidth()});
+      this._overlayRef.updateSize({width: this._getWidth()});
```

Comment:
while `updateSize` does any in `number | string` as its type for width, it ends up just coercing the value provided into `{$value}px`.  So if `auto` was provided as the value it would end up as `autopx`.

Reply:
It only does that if the value is a `number`

```
export function coerceCssPixelValue(value: any): string {
  if (value == null) {
    return '';
  }

  return typeof value === 'string' ? value : `${value}px`;
}
```

Reply:
Ahh yes, I glanced too quickly when I read it.